### PR TITLE
MGMT-9108: Remove V1 GetPresignedForClusterFiles (GET)

### PR DESCRIPTION
**Pull request by nmagnezi**
## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @filanov 
/cc @danielerez 

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -4438,12 +4439,7 @@ func (b *bareMetalInventory) V2GetPresignedForClusterFiles(ctx context.Context,
 }
 
 func (b *bareMetalInventory) GetPresignedForClusterFiles(ctx context.Context, params installer.GetPresignedForClusterFilesParams) middleware.Responder {
-	presigned, err := b.getPresignedForClusterFiles(ctx, &params.ClusterID, params.FileName, params.AdditionalName, params.HostID, params.LogsType)
-	if err != nil {
-		return err
-	}
-
-	return installer.NewGetPresignedForClusterFilesOK().WithPayload(presigned)
+	return common.GenerateErrorResponder(errors.New(APINotImplemented))
```
Comment (filanov) [inline internal/bminventory/inventory.go:4442]: Isn't that 501? https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501

Comment (nmagnezi) [reply internal/bminventory/inventory.go]: Yes, Done.

File: subsystem/authz_test.go [hunk 2]
```diff
@@ -203,9 +203,9 @@ var _ = Describe("Make sure that sensitive files are accessible only by owners o
 		})
 		it = fmt.Sprintf("Should not allow read-only-admins  to download '%v' via downloads/files-presigned endpoint", fileName)
 		It(it, func() {
-			_, err := readOnlyAdminUserBMClient.Installer.GetPresignedForClusterFiles(ctx, &installer.GetPresignedForClusterFilesParams{ClusterID: clusterID, FileName: fileName})
+			_, err := readOnlyAdminUserBMClient.Installer.V2GetPresignedForClusterFiles(ctx, &installer.V2GetPresignedForClusterFilesParams{ClusterID: clusterID, FileName: fileName})
 			Expect(err).To(HaveOccurred())
-			Expect(reflect.TypeOf(err)).Should(Equal(reflect.TypeOf(installer.NewGetPresignedForClusterFilesForbidden())))
+			Expect(reflect.TypeOf(err)).Should(Equal(reflect.TypeOf(installer.NewV2GetPresignedForClusterFilesForbidden())))
```
Comment (carbonin) [inline subsystem/authz_test.go:208]: While we're here can you make this assertion a bit more helpful?
This is not a good failure message:
```
/go/src/github.com/openshift/assisted-service/subsystem/authz_test.go:205
Expected
    <*reflect.rtype | 0x1a8efc0>: {size: 0x8, ptrdata: 0x8, hash: 3563199731, tflag: 9, align: 8, fieldAlign: 8, kind: 54, equal: 0x404700, gcdata: 1, str: 157180, ptrToThis: 0}
to equal
    <*reflect.rtype | 0x1acf4a0>: {size: 0x8, ptrdata: 0x8, hash: 2194802269, tflag: 9, align: 8, fieldAlign: 8, kind: 54, equal: 0x404700, gcdata: 1, str: 608164, ptrToThis: 0}
/go/src/github.com/openshift/assisted-service/subsystem/authz_test.go:208
```

Comment (carbonin) [reply subsystem/authz_test.go:208]: There's a gomega built-in matcher `BeAssignableToTypeOf` that I think would work here.

File: subsystem/authz_test.go [hunk 3]
```diff
@@ -217,8 +217,8 @@ var _ = Describe("Make sure that sensitive files are accessible only by owners o
 		})
 		it = fmt.Sprintf("Should allow cluster users to download '%v' via downloads/files-presigned endpoint", fileName)
 		It(it, func() {
-			_, err := userBMClient.Installer.GetPresignedForClusterFiles(ctx, &installer.GetPresignedForClusterFilesParams{ClusterID: clusterID, FileName: fileName})
-			Expect(reflect.TypeOf(err)).ShouldNot(Equal(reflect.TypeOf(installer.NewGetPresignedForClusterFilesForbidden())))
+			_, err := userBMClient.Installer.V2GetPresignedForClusterFiles(ctx, &installer.V2GetPresignedForClusterFilesParams{ClusterID: clusterID, FileName: fileName})
+			Expect(reflect.TypeOf(err)).ShouldNot(Equal(reflect.TypeOf(installer.NewV2GetPresignedForClusterFilesForbidden())))
```
Comment (carbonin) [inline subsystem/authz_test.go:221]: This one should probably change too.

**Comment by nmagnezi** [comment]
/hold
fixing subsystem tests

File: internal/ignition/ignition.go [hunk 4]
```diff
@@ -367,14 +367,14 @@ const secondDayWorkerIgnitionFormat = `{
...
 const tempNMConnectionsDir = "/etc/assisted/network"
 
-var fileNames = [...]string{
-	"bootstrap.ign",
-	masterIgn,
-	"metadata.json",
-	workerIgn,
-	"kubeconfig-noingress",
+var FileNames = [...]string{
+	//"bootstrap.ign",
```
Comment (filanov) [inline internal/ignition/ignition.go:371]: if not needed please remove

Comment (nmagnezi) [reply internal/ignition/ignition.go:371]: Still WIP so I kept the `hold` label. I'll post a comment when it is ready for review.

**Comment by nmagnezi** [comment]
> New changes are detected. LGTM label has been removed.

Fixed a lint issue.

**Review by carbonin — APPROVED** [review]

**Pull request merged**